lib/ibm_watson/watson_service.rb

```diff
@@ -39,6 +39,7 @@ def initialize(vars)
     @password = nil
     @token_manager = nil
     @temp_headers = nil
+    @icp_prefix = vars[:password].start_with?("icp-") ? true : false unless vars[:password].nil?
```

Comment:
The `unless` conditional still causes `@icp_prefix` to be undefined if there is no password present. I believe that `@icp_prefix = vars[:password]&.start_with?("icp-") ? true : false` should work. The `&` symbol calls the `start_with?` method if the method exists for the object (I think, but it should work).